db/src/schema.rs

```diff
@@ -52,6 +55,135 @@ fn validate_schema_map(entid_map: &EntidMap, schema_map: &SchemaMap) -> Result<(
     Ok(())
 }
 
+#[derive(Clone,Debug,Default,Eq,Hash,Ord,PartialOrd,PartialEq)]
+pub struct AttributeBuilder {
+    value_type: Option<ValueType>,
+    multival: Option<bool>,
+    unique_value: Option<bool>,
+    unique_identity: Option<bool>,
+    index: Option<bool>,
+    fulltext: Option<bool>,
+    component: Option<bool>,
+}
+
+impl AttributeBuilder {
+    pub fn value_type<'a>(&'a mut self, value_type: ValueType) -> &'a mut Self {
```

Comment:
This is not being consumed in a fluent interface (like `Builder::new().thing1().thing2().build()`), but in a loop that grabs the builder reference, does a thing, and then finally builds everything at the end.  It's equivalent to consume `self` and update the `&mut Builder` in the map entry, I guess.